Add Programmatic Access to Choreo Insights API doc (#437)
wso2/docs-choreo-dev: en/docs/insights/programmatic-access-choreo-insights-api.md

# Programmatic Access to Insights API

> The insights page in the Choreo Console fetches data from the Insights API exposed over the Internet. Occasionally, some users might need to fetch these data from external systems for various use cases. One such use case can be to implement their own dashboards based on these data.

Reviewer: ```suggestion
The Insights page in Choreo Console fetches data from the Insights API exposed over the Internet. Occasionally, you may need to fetch data from external systems and implement custom dashboards based on this data.
```

Choreo provides a secure and user-friendly method to achieve this requirement. The steps are outlined below.

> ## Become a Member of the Choreo System Organization

Reviewer: ```suggestion
Become a member of the Choreo System organization.
```

1. Sign in to [https://devportal.choreo.dev](https://devportal.choreo.dev) using your credentials.

> 2. Visit [https://devportal.choreo.dev/choreosystem](https://devportal.choreo.dev/choreosystem) to access the developer portal of Choreo System organization.

Reviewer: ```suggestion
Access the developer portal of the `Choreo System` organization via [https://devportal.choreo.dev/choreosystem](https://devportal.choreo.dev/choreosystem).
```

>     If you are accessing the Choreo System organization’s developer portal for the first time, you will need request access to the organization by clicking on the **Request Access** button.

Reviewer: ```suggestion
    !!! info
        If you are accessing the Developer Portal of the `Choreo System` organization for the first time, request access by clicking **Request Access**.
        ![Request-Access](../assets/img/insights/request-access.png){.cInlineImage-full}
 
        Once you request access, you are  automatically added as a member of the `Choreo System` organization.
 
        Verify this by checking whether the `Choreo System` organization appears in your list of organizations.
 
        ![Choreo-System-Home](../assets/img/insights/choreo-system-home.png){.cInlineImage-full}
```

    ![Request-Access](../assets/img/insights/request-access.png){.cInlineImage-full}

3. Once access has been requested, you will be automatically added as a member of the Choreo System organization.

    Verify this by inspecting the organization drop-down(Choreo System organization should be visible).

    ![Choreo-System-Home](../assets/img/insights/choreo-system-home.png){.cInlineImage-full}

> ## Create an Application & Generate Keys

Reviewer: ```suggestion
## Create an application and generate keys
```

Reviewer: Have an introductory sentence before the steps.

e.g., To create an application and generate keys, follow the steps below:

> 1. Navigate to the Applications' menu to create a new application in the Choreo System organization.

Reviewer: ```suggestion
1. Click on the **Applications** link to create a new application in the Choreo System organization.
```

>     ![Create-Application](../assets/img/insights/create-application.png){.cInlineImage-full}

Reviewer: Align each image within a bullet point. This comment applies to all the images.

> 2. Provide a suitable name and create the application.

Reviewer: Let's use "Enter" instead of "Provide"?

    ![Create-Application-Wizard](../assets/img/insights/create-application-wizard.png){.cInlineImage-half}

3. Subscribe to the **Insights** API using the newly created application.

>     ![Subscribe-API](../assets/img/insights/subscribe-api.png){.cInlineImage-half}

Reviewer: Align the image within a bullet point. This comment applies to the next two images as well.

Reviewer: Shall we describe how it is done?

e.g., "To subscribe to the new API via the newly created application, click **Subscriptions** in the left navigator, and then click **Add** for the **Insights** API."

> 4. Generate credentials for the application. This will generate a consumer key and secret for the created application.

Reviewer: ```suggestion
To generate credentials for the application, click  **Production Keys** -> **OAuth 2.0 Tokens** in the left navigator, and then click **Generate Credentials**.  This generates a consumer key and secret for the created application.
```

    ![Generate-Credentials](../assets/img/insights/generate-credentials.png){.cInlineImage-full}

> 5. Provide a suitable token expiry time and generate a token.

Reviewer: ```suggestion
Specify a suitable token expiry time and generate a token by clicking **Generate Test Token**.
```

    ![Generate-Test-Token](../assets/img/insights/generate-test-token.png){.cInlineImage-full}

> 6. This token can then be used to access the Insights API exposed via [https://choreocontrolplane.choreo.dev/93tu/insights/1.0.0/query-api](https://choreocontrolplane.choreo.dev/93tu/insights/1.0.0/query-api)

Reviewer: This should not be a step because it does not ask the user to carry out a specific action. Let's align it below Bullet point 5..

    A sample curl command is shown below.

    ```bash

>     curl --location --request POST 'https://choreocontrolplane.choreo.dev/93tu/insights/1.0.0/query-api' \

Reviewer: Is this relevant only to the step about generating a test token? If so, indent this text so that it is aligned with the bullet point.

    --header 'Content-Type: application/json' \
    --header 'Authorization: Bearer <TOKEN_HERE>' \
    --data-raw '{"query":"query ($org: OrgFilter!) {listEnvironments(org: $org){id\n name}}","variables":{"org":{"orgId":"<ORG_UUID_HERE>"}}}'
    ```

>     In case of token expiry, a new token can be generated programmatically by talking to the token-endpoint using the client credentials grant.

Reviewer: ```suggestion
When the token expires, you can generate a new token programmatically by talking to the token endpoint via the client credentials grant.
```

>     In case of a token theft, the relevant token can be revoked by talking to the revoke-endpoint.

Reviewer: ```suggestion
If a token theft occurs, the relevant token can be revoked by communicating with the revoke-endpoint.

# Programmatic Access to Insights API

> The **Insights** page in the Choreo Console fetches data from the Insights API exposed over the Internet. However, in some scenarios, you may need to fetch data from external systems and implement custom dashboards to display this data.

Reviewer 2: Doesn't seem correct to say "The **Insights** page in the Choreo Console fetches data. . . ". The term page doesn't seem appropriate I feel.

Author: We have a separate page for insights in the console. So why it is wrong to use that term? Or else do you have any suggestion for this?

Reviewer 2: When we have it as "The Insights page fetches data from the Insights API. . ." it is not quite right because the Insights page is only displaying the data. Not really fetching the data.
We would have to say something similar  
The Insights page in Choreo displays data fetched from the Insights API exposed over the internet.

Author: Fixed with [15541b0](https://github.com/wso2/docs-choreo-dev/pull/437/commits/15541b0254dadedf8213c084c17bfd6498b6053d)

> Choreo provides a secure and user-friendly method to achieve this requirement. The steps are outlined below.

Reviewer 2: ```suggestion
Choreo provides a secure and user-friendly method to achieve this requirement. The steps are outlined below:
```

Author: Fixed with [54aec2f](https://github.com/wso2/docs-choreo-dev/pull/437/commits/54aec2f020296f46f683ab402e4183683e433051)

## Become a member of the Choreo System organization

1. Sign in to [https://devportal.choreo.dev](https://devportal.choreo.dev) using your credentials.

> 2. Access the developer portal of the `Choreo System` organization via [https://devportal.choreo.dev/choreosystem](https://devportal.choreo.dev/choreosystem).

Reviewer 2: We need to use title caps for the term "Developer Portal".

Author: Fixed with [54aec2f](https://github.com/wso2/docs-choreo-dev/pull/437/commits/54aec2f020296f46f683ab402e4183683e433051)

> 3. Request Access to the `Choreo System` organization by clicking **Request Access**.

Reviewer 2: Unnecessary capitalization on the term access.

Shall we say
"Click **Request Access** to get access to the `Choreo System` organization."
or 
"To get access to the `Choreo System` organization, click **Request Access**.

Author: Fixed with [54aec2f](https://github.com/wso2/docs-choreo-dev/pull/437/commits/54aec2f020296f46f683ab402e4183683e433051)
